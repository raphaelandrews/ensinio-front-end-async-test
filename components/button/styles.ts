import styled from 'styled-components';

export const Button = styled.button`
    position: relative;
    display: flex;
    align-items: center;
    gap: .75rem;
    font-size: .9375rem;
    font-weight: 600;
    color: ${({ theme }) => theme.colors.background};
    line-height: 105%;
`

export const OutlineButton = styled(Button)`
    padding: .75rem 1.5rem;
    border: 1px solid hsla(0, 0%, 100%, .5);
    border-radius: 5rem;
`