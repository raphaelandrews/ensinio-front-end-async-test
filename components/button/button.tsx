import { ReactNode } from 'react';

import * as S from './styles';

interface ButtonProps {
    children: ReactNode;
}

export const Button = ({ children }: ButtonProps) => {
    return (
        <S.Button>
            {children}
        </S.Button>
    );
}

export const OutlineButton = ({ children }: ButtonProps) => {
    return (
        <S.OutlineButton>
            {children}
        </S.OutlineButton>
    );
}