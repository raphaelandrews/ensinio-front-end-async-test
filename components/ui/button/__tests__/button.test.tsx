import { fireEvent, render, screen } from '@testing-library/react';
import { Button, OutlineButton } from "@/components/ui/button/button"

import Providers from '@/app/providers';

describe('Button', () => {
    it('renders Button component with correct children', () => {
        render(
            <Providers>
                <Button>Click Me </Button>
            </Providers>
        );

        expect(screen.getByText('Click Me'))
    });

    it('calls onClick handler when Button is clicked', () => {
        const onClickMock = jest.fn();

        render(
            <Providers>
                <Button onClick={onClickMock}>Click Me</Button>
            </Providers>
        );

        fireEvent.click(screen.getByText('Click Me'));
        expect(onClickMock).toHaveBeenCalledTimes(1);
    });

    it('calls onMouseEnter handler when Button is hovered over', () => {
        const onMouseEnterMock = jest.fn();

        render(
            <Providers>
                <Button onMouseEnter={onMouseEnterMock}>Hover Me</Button>
            </Providers>
        );

        fireEvent.mouseEnter(screen.getByText('Hover Me'));
        expect(onMouseEnterMock).toHaveBeenCalledTimes(1);
    });


    it('renders OutlineButton component with correct children', () => {
        render(
            <Providers>
                <OutlineButton>Outline Button</OutlineButton>
            </Providers>
        );

        expect(screen.getByText('Outline Button'));
    });

    it('matches Button component snapshot', () => {
        const { container } =
            render(
                <Providers>
                    <Button >Click Me</Button>
                </Providers>
            );

        expect(container).toMatchSnapshot();
    });

    it('matches OutlineButton component snapshot', () => {
        const { container } =
            render(
                <Providers>
                    <Button >Click Me</Button>
                </Providers>
            );

        expect(container).toMatchSnapshot();
    });
});