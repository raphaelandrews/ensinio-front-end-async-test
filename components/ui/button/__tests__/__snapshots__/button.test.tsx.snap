// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button matches Button component snapshot 1`] = `
<div>
  <button
    class="sc-bdnyFh hRlNnM"
  >
    Click Me
  </button>
</div>
`;

exports[`Button matches OutlineButton component snapshot 1`] = `
<div>
  <button
    class="sc-bdnyFh hRlNnM"
  >
    Click Me
  </button>
</div>
`;
